projects/packages/forms/src/blocks/shared/hooks/use-container-id.js

```diff
@@ -0,0 +1,43 @@
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Custom hook to retrieve the container ID of a 'jetpack/contact-form' block.
+ *
+ * The container could be places anywhere in the DOM, so we need to check
+ * if the formClientId is in the parent form. ( since the document could have multiple forms )
+ *
+ * @param {string} formClientId - The client ID of the parent 'jetpack/contact-form' block.
+ * @return {string|null} The client ID of the 'jetpack/step-container' block, or null if not found.
+ */
+export default function useContainerId( formClientId ) {
```

Comment:
Nit: The hook's name is a bit generic, it's not clear what type of container or id it returns 

```suggestion
export default function useStepContainerClientId( formClientId ) {
```